calendar-bundle/src/Resources/contao/classes/Events.php

```diff
@@ -202,7 +202,6 @@ protected function getAllEvents($arrCalendars, $intStart, $intEnd, $blnFeatured
 	 */
 	protected function addEvent($objEvents, $intStart, $intEnd, $intBegin, $intLimit, $intCalendar)
 	{
-		/** @var PageModel $objPage */
 		global $objPage;
```

Comment:
How does the IDE now know about the type of `global $objPage`?

Reply:
I don't really know how it knows, I just know that is knows. 😄 

<img width="554" alt="" src="https://user-images.githubusercontent.com/1192057/87934739-eb989180-ca8f-11ea-8282-1c5c37a6a241.png">

Reply:
I think this is a false positive. If I remove the annotation, PHPStorm won't complain, but it also won't give me any autocompletion or type info anymore.

Reply:
That's really strange. Seems I have a plugin that can do it:

<img width="972" alt="" src="https://user-images.githubusercontent.com/1192057/87939988-2a7f1500-ca99-11ea-8458-ba26f26c65b5.png">

Reply:
I have reverted the `global $objPage` changes in 12840066dfbaaede838eab0bd484133249a3db54, although I still wonder why my PhpStorm was able to infer the type and yours was not. 🤷🏻‍♂️

Reply:
In your last image `$objUploader` can be inferred from `$objUploader = new FileUpload()`.

I was referring to the `global $objPage` annotations before.

![image](https://user-images.githubusercontent.com/5305677/87941427-6fa44680-ca9b-11ea-8b93-e0f719bc276c.png)

vs

![image](https://user-images.githubusercontent.com/5305677/87941349-4edbf100-ca9b-11ea-8480-00a83fd6e0dc.png)